tests/logging/main.c

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#define PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP \
```

Comment:
Is this used anywhere?

Reply:
host/traceh.c uses `OE_H_MUTEX_INITIALIZER` which is defined in `hostthread.h` and which in turn uses the definition of `PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP`. The latter I belive is defined in `pthread.h`. I thought it would be ok to define it here like this for the purposes of this test.